app/utils/autoscale.ts

```diff
@@ -17,21 +17,50 @@ const systemFunc = [
     "GetSample", "GetSamplePeriod", "Count", "HistoryBeginTime", "GetSamplePercent",
 ];
 
+const keywords = [
+    "double", "doubleVec", "doubleVecList", "string", "timestamp",
+];
+
+const timeInterval = [
+    "TimeInterval_Zero",
+    "TimeInterval_100ns",
+    "TimeInterval_Microsecond",
+    "TimeInterval_Millisecond",
+    "TimeInterval_Second",
+    "TimeInterval_Minute",
+    "TimeInterval_Hour",
+    "TimeInterval_Day",
+    "TimeInterval_Week",
+    "TimeInterval_Year",
+];
+
 // tslint:disable-next-line:max-line-length
-const keywordRegex = /\$CPUPercent|\$WallClockSeconds|\$MemoryBytes|\$DiskBytes|\$DiskReadBytes\$DiskWriteBytes|\$DiskReadOps|\$DiskWriteOps|\$NetworkInBytes|\$NetworkOutBytes|\$SampleNodeCount|\$ActiveTasks|\$RunningTasks|\$PendingTasks|\$SucceededTasks|\$FailedTasks|\$CurrentDedicated|\$TargetDedicated/;
-const functionsRegex = /avg|len|lg|ln|log|max|min|norm|percentile|rand|range|std|stop|sum|time|val/;
-const obtainRegex = /GetSample|GetSamplePeriod|Count|HistoryBeginTime|GetSamplePercent/;
-const atomRegex = /true|false|null/;
+const keywordRegex = /\$\b(CPUPercent|WallClockSeconds|MemoryBytes|DiskBytes|DiskReadBytes\$DiskWriteBytes|DiskReadOps|DiskWriteOps|NetworkInBytes|NetworkOutBytes|SampleNodeCount|ActiveTasks|RunningTasks|PendingTasks|SucceededTasks|FailedTasks|CurrentDedicated|TargetDedicated|NodeDeallocationOption)\b/;
```

Comment:
Can't you do that for all the following?

```
const keywordRegex = new Regex(variables.join("|"))?
```

you might need to escape `$` too